io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -754,6 +774,58 @@ impl LvsLvol for Lvol {
 
         info!("{:?}: creating snapshot '{}'", self, snapshot_name);
     }
+
+    /// Blobstore Common Callback function.
+    extern "C" fn blob_op_complete_cb(
+        arg: *mut c_void,
+        blob: *mut spdk_blob,
+        errno: i32,
+    ) {
+        let s = unsafe {
+            Box::from_raw(arg as *mut oneshot::Sender<(*mut spdk_blob, i32)>)
+        };
+        if errno != 0 {
+            error!("spdk_bs_iter_first failed errno {}", errno);
+        }
+        s.send((blob, errno)).ok();
+    }
+
+    /// Get the first spdk_blob from the Lvol Blobstor.
+    fn bs_iter_first(&self) -> *mut spdk_blob {
+        self.as_inner_ref().blob
+    }
+
+    /// Get the next spdk_blob from the current blob.
+    async fn bs_iter_next(
+        &self,
+        curr_blob: *mut spdk_blob,
+    ) -> Option<*mut spdk_blob> {
+        let (s, r) = oneshot::channel::<(*mut spdk_blob, i32)>();
+        unsafe {
+            spdk_bs_iter_next(
+                self.lvs().blob_store(),
+                curr_blob,
+                Some(Self::blob_op_complete_cb),
+                cb_arg(s),
+            )
+        };
+        match r.await {
+            Ok((blob, _err)) => Some(blob),
```

Comment:
What is _err here, sounds like it could be an error, if so, should we handle it or we don't care?

Reply:
Currently the error is not handled directly. The blob will be NULL if there will be any error and next iterator will be exit.